Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -1252,6 +1252,114 @@ WebIDL::ExceptionOr<JS::GCPtr<PendingResponse>> http_redirect_fetch(JS::Realm& r
     return main_fetch(realm, fetch_params, recursive);
 }
 
+// https://fetch.spec.whatwg.org/#network-partition-key
+struct NetworkPartitionKey {
+    HTML::Origin top_level_origin;
+    // FIXME: See https://github.com/whatwg/fetch/issues/1035
+    void* second_key = nullptr;
```

Comment:
This the document origin in other browsers, so we could do that here (eventually - not sure we track that info now)

Reply:
We have a few things about Environments and origins that we do a bit cowboy, so I doubt we do have that info on  hand properly. The spec says we're allowed to not use the second key anyway (even though everyone does :) ). I'll add a comment with that info though.